pytest_bdd/steps.py

```diff
@@ -183,7 +134,9 @@ def lazy_step_func():
         if converters:
             step_func.converters = lazy_step_func.converters = converters
 
-        lazy_step_func = pytest.fixture(scope=scope)(lazy_step_func)
+        step_func.target_fixture = lazy_step_func.target_fixture = target_fixture
+
+        lazy_step_func = pytest.fixture()(lazy_step_func)
         setattr(get_caller_module(), get_step_fixture_name(parsed_step_name, step_type), lazy_step_func)
```

Comment:
and IIRC this line is to make pytest discover the fixture

Reply:
yes. So you can specify "given foo" in the conftest and override it in your local file